ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/General.hs

```diff
@@ -392,13 +400,40 @@ prop_general ::
   forall blk.
      ( Condense blk
      , Eq blk
-     , HasHeader blk
      , RunNode blk
      )
   => PropGeneralArgs blk
   -> TestOutput blk
   -> Property
-prop_general pga testOutput =
+prop_general = prop_general_internal Sync
+
+-- | Like 'prop_general' but instead assuming /semi-synchrony/
+--
+-- For now, this simply disables a few 'Property's that depend on synchrony.
+prop_general_semisync ::
+  forall blk.
+     ( Condense blk
+     , Eq blk
+     , RunNode blk
+     )
+  => PropGeneralArgs blk
+  -> TestOutput blk
+  -> Property
+prop_general_semisync = prop_general_internal SemiSync
+
+data Synchronicity = SemiSync | Sync
```

Comment:
A comment briefly explaining what you mean by synchrony and semi-synchrony would be helpful.

Reply:
Good catch. Done! Thanks.